plugin/pkg/scheduler/scheduler.go

```diff
@@ -89,6 +92,12 @@ func (s *Scheduler) Run() {
 
 func (s *Scheduler) scheduleOne() {
 	pod := s.config.NextPod()
+
+	// If the assumed pods store contains this pod there is a conflict,
+	// since the store can't contain a pod without a host, and NextPod
+	// only gives us pods without hosts.
+	s.config.Modeler.ForgetAssumedPod(pod)
```

Comment:
This doesn't strike me as the right mechanism for avoiding machines that Kubelet thinks are incompatible with the pod. We should have some more principled approach like explicitly tracking such machines on a per-pod basis and avoiding them for some period of time. 

If the only reason not to make the change @beeps made is to avoid binding again to the same machine, then I think the change is reasonable. It doesn't seem *incorrect* AFAICT--if a pod shows up in the queue again, we know our "assumption" was wrong, so deleting it from assumed seems reasonable.